### ISSUE TITLE
Setting descriptor directly from method call causes a type issue

### ISSUE THREAD

**Issue by Bluenix2** [microsoft/pyright#3121]
## Describe the bug

I have descriptor that I set to the result of a method call (more specifically `contextlib.ExitStack.enter_context()`) but Pyright complains that the result of the method has incompatible types with the attribute/descriptor.

## Reproduction of the bug

The following code demonstrates the bug:
```python
class GenericDescriptor(Generic[T]):
    value: T

    def __get__(self, instance: Optional[object], cls: Type[object]) -> T:
        return self.value

    def __set__(self, instance: object, value: T) -> None:
        self.value = value


class Test:
    abc: GenericDescriptor[str] = GenericDescriptor()
    stack: ExitStack

    def test(self, value: ContextManager[str]) -> None:
        self.abc = self.stack.enter_context(value)
```
```
Cannot assign member "abc" for type "Test"
  "GenericDescriptor[str]" is incompatible with "str"
```
```
Argument of type "ContextManager[str]" cannot be assigned to parameter "cm" of type "AbstractContextManager" in function "enter_context"
  TypeVar "_T_co@ContextManager" is covariant
    Type "str" cannot be assigned to type "GenericDescriptor[str]"
```

### Workaround

I am able to work around this error by first assigining the method return to a variable, then set the descriptor to this variable:
```python
    def test(self, value: ContextManager[str]) -> None:
        abc = self.stack.enter_context(value)
        self.abc = abc
```

## Installation information
Pylance is disabled on my Visual Studio Code installation.

```
Released on 19/01/2016, 16:03:11
Last updated 08/02/2022, 22:04:46
Identifier ms-python.python
```
```
Released on 22/03/2019, 04:10:23
Last updated 26/02/2022, 16:33:42
Identifier ms-pyright.pyright
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in 1.1.226, which I just published. It will also be included in the next release of pylance.
